dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/jfapchannel/richclient/framework/impl/RichClientTransportFactory.java

```diff
@@ -79,31 +85,44 @@ public NetworkConnectionFactory getOutboundNetworkConnectionFactoryByName(String
 
         try
         {
-            // Get the virtual connection factory from the channel framework using the chain name
-
-            VirtualConnectionFactory vcFactory = CommsClientServiceFacade.getChannelFramewrok().getOutboundVCFactory(chainName);
+        	// Get the virtual connection factory from the channel framework using the chain name
+        	if(CommsOutboundChain.getChainDetails(chainName) != null) {
+        		if(CommsOutboundChain.getChainDetails(chainName).useNetty()) {
+        			// If Netty, create new Netty channel factory
+        			// TODO: Consult with team. Getting error difference cause channelfw fails for testSSLFeatureUpdate
+        			// in com.ibm.ws.messaging.open_comms_fat because SSL chain failed to init properly due to no SSL Options
+        			// Check what to do here appropriately
+        			CommsOutboundChain chain = CommsOutboundChain.getChainDetails(chainName);
+        			boolean usingSSL = chain.isSSL();
+        			if(usingSSL && chain.getSslOptions() == null)
+        				throw new InvalidChainNameException("Chain configuration not found in framework, " + chainName);
+        			connFactory = new NettyNetworkConnectionFactory(chainName, chain.getTcpOptions(), usingSSL?chain.getSslOptions():null, usingSSL?chain.getTlsProvider():null);
+        		}else {
+        			VirtualConnectionFactory vcFactory = CommsClientServiceFacade.getChannelFramework().getOutboundVCFactory(chainName);
+        			connFactory = new CFWNetworkConnectionFactory(vcFactory);
+        		}
+        	}
 
-            connFactory = new CFWNetworkConnectionFactory(vcFactory);
 
         } catch (com.ibm.wsspi.channelfw.exception.ChannelException e) {
 
-            FFDCFilter.processException(e, CLASS_NAME + ".getOutboundNetworkConnectionFactoryByName",
-                                        JFapChannelConstants.RICHCLIENTTRANSPORTFACT_GETCONNFACBYNAME_01,
-                                        new Object[] { this, chainName });
+        	FFDCFilter.processException(e, CLASS_NAME + ".getOutboundNetworkConnectionFactoryByName",
```

Comment:
I'm not entirely sure what was the decision behind this since this is actually part of the JMS original code. As for OVC I think it means Outbound Virtual Connection but not 100% sure